packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -693,6 +737,40 @@ function commitRoot(root: FiberRoot, finishedWork: Fiber): void {
     legacyErrorBoundariesThatAlreadyFailed = null;
   }
   onCommit(root, earliestRemainingTimeAfterCommit);
+
+  if (enableInteractionTracking) {
+    ((__interactionsRef: any): InteractionsRef).current = ((prevInteractions: any): Set<
+      Interaction,
+    >);
+
+    if (enableInteractionTrackingObserver) {
+      if (((root.memoizedInteractions: any): Set<Interaction>).size > 0) {
+        // Interaction threads are unique per root and expiration time.
+        const threadID =
+          committedExpirationTime * 1000 + root.interactionThreadID;
+        ((__subscribers: any): Subscribers).forEach(observer =>
+          observer.onWorkStopped(
```

Comment:
Related to my concerns in https://github.com/facebook/react/pull/13234#discussion_r209807981, I think we need to be very confident in user-level code to call it from this place in the scheduler. Since it's not inside the render phase or one of protected places in the commit phase, there's a risk that any error in it (and folks who want to use it are humans and may make mistakes) can mess up the scheduler state and cause infinite loops etc.

Or we could wrap any user code call like this in a try/catch.

Reply:
Great point. Although I'm not sure what the expected error recover behavior really should be in this case. I'll take a stab at it– with some new tests– and we can discuss it from there.